model_analyzer/analyzer.py

```diff
@@ -266,25 +300,45 @@ def _profile(self, perf_analyzer, dcgm_monitor):
 
         # Insert all records into aggregator and get aggregated DCGM records
         record_aggregator = RecordAggregator()
-        for record in perf_records + dcgm_records:
+        for record in dcgm_records:
             record_aggregator.insert(record)
-        return record_aggregator.aggregate()
 
-    def _create_output_table(self, aggregation_tag='Max'):
+        def gpu_device_id(record):
+            return record.device().device_id()
+
+        records_groupby_gpu = {}
+        for tag in self.dcgm_tags:
+            records_groupby_gpu[tag] = record_aggregator.groupby(
+                tag, gpu_device_id)
```

Comment:
For single use, maybe a lambda would be preferable to nested def. 
```
for tag in self.dcgm_tags:
    records_groupby_gpu[tag] = record_aggregator.groupby(
         tag, (lambda record : record.device().device_id()))
```
I suppose this is stylistic. I think nested defs are recommended when there's repeated use but in this case it is use and throw right?

Reply:
Right, good suggestion!